core/savings.go

```diff
@@ -125,28 +105,20 @@ func (s *Savings) updatePrices(p publisher) (float64, float64) {
 	return gridPrice, feedinPrice
 }
 
-func (s *Savings) Update(p publisher, gridPower, pvPower, batteryPower, chargePower float64) {
+// Update savings calculation and return grid/green energy added since last update
+func (s *Savings) Update(p publisher, deltaCharged, deltaGrid float64) {
 	gridPrice, feedinPrice := s.updatePrices(p)
-	defer func() { s.updated = s.clock.Now() }()
 
-	// no charging, no need to update
-	if chargePower == 0 {
-		return
-	}
+	addedSelfConsumption := deltaCharged - deltaGrid
```

Comment:
Du änderst hier das Berechnungsmodell. Es reicht nicht einfach nur das Delta an Netzbezug zu betrachten.
Das aktuelle Modell nimmt den gesamten Energiemix im Haus als Grundlage. Hier ist dokumentiert wie die Berechnung funktioniert: https://docs.evcc.io/docs/guides/setup/#ersparnisberechnung

Reply:
> Die direkte PV-Nutzung und die vom Batteriesystem bereitgestellte Energie werden bei dieser Berechnung gleich behandelt. evcc geht davon aus, dass der Hausspeicher primär zur Speicherung von überschüssigem, selbstproduzierten Sonnenstrom verwendet wird. Erfüllt der Speicher auch netzdienliche Leistungen (bspw. Sonnen) oder führt eine Kalibrierung mit Netzstrom durch, ist diese Annahme nicht immer korrekt. Batterieverluste durch die Umwandlung werden nicht berücksichtigt.

Wo siehst Du den Unterschied? Ich behandelte nur grid als grid. Der Rest ist grün wie's in der Doku steht.